server/src/main/java/io/druid/client/selector/RandomServerSelectorStrategy.java

```diff
@@ -34,4 +38,14 @@ public QueryableDruidServer pick(Set<QueryableDruidServer> servers, DataSegment
   {
     return Iterators.get(servers.iterator(), random.nextInt(servers.size()));
   }
+
+  @Override
+  public List<QueryableDruidServer> pick(
+      Set<QueryableDruidServer> servers, DataSegment segment, int numServersToPick
+  )
+  {
+    List<QueryableDruidServer> list = Lists.newArrayList(servers);
+    Collections.shuffle(list);
+    return list.subList(0, numServersToPick);
```

Comment:
This will break if `list.size` < `numServersToPick` can this pick up to this number?